packages/core/admin/ee/server/services/review-workflows/stages.js

```diff
@@ -62,50 +65,32 @@ module.exports = ({ strapi }) => {
         const defaultWorkflow = await getDefaultWorkflow({ strapi });
         const newStages = await this.createMany(created, { fields: ['id'] });
         const stagesIds = stages.map((stage) => stage.id ?? [...newStages].shift().id);
+        const contentTypes = getContentTypeUIDsWithActivatedReviewWorkflows(strapi.contentTypes);
 
         await mapAsync(updated, (stage) => this.update(stage.id, stage));
 
-        const entitiesToMove = [];
         await mapAsync(deleted, async (stage) => {
-          // Find any entities related to this stage
-          const stageInfo = await this.findById(stage.id, {
-            populate: ['related'],
-          });
-
-          if (!stageInfo?.related?.length) {
-            // If there are no related entities, just delete the stage
-            return this.delete(stage.id);
-          }
-
-          // This stage has related entities that need to be moved to their
-          // target stage
           // Find the nearest stage in the workflow and newly created stages
           // that is not deleted, prioritizing the previous stages
-          const targetStageId = findNearestMatchingStageID(
+          const nearestStage = findNearestMatchingStage(
             [...defaultWorkflow.stages, ...newStages],
             defaultWorkflow.stages.findIndex((s) => s.id === stage.id),
             (targetStage) => {
               return !deleted.find((s) => s.id === targetStage.id);
             }
           );
 
-          // Keep track of the entities to move and their target stages
-          entitiesToMove.push(...stageInfo.related.map((entity) => ({ ...entity, targetStageId })));
+          // Assign the new stage to entities that had the deleted stage
+          await mapAsync(contentTypes, (contentTypeUID) =>
```

Comment:
Not a big fan of nested loops, but it should not be massive here, and maybe we can try to find other ways to simplify this.